### ISSUE TITLE
Migrate PWA Studio to magento Organization

### ISSUE THREAD

**Issue by supernova-at** [magento/pwa-studio#1472]
https://help.github.com/en/articles/transferring-a-repository#whats-transferred-with-a-repository
https://help.github.com/en/articles/transferring-a-repository#transferring-a-repository-owned-by-your-organization

# Plan of action:

🔁 denotes an item in progress

## Before the Move

- [x] Prune branches of `pwa-studio` repository  
- [x] Identify CICD integrations that may potentially break 
**Done:** https://github.com/magento-research/pwa-studio/issues/1472#issuecomment-516067641.
- [x] Notify the community (Slack, Community Sync)
- [x] Create a full list of outside collaborators to add to the Community Contributors team. 
**Done:** https://github.com/magento-research/pwa-studio/issues/1472#issuecomment-515589081.
- [x] Record project collaborators' permission set (will not transfer to the new repo automatically). **Done:** https://github.com/magento-research/pwa-studio/issues/1472#issuecomment-516133700.
- [x] Create a full repository copy using [GitHub Mirror approach](https://help.github.com/en/articles/duplicating-a-repository) 
**Done:** https://github.com/magento-research/pwa-studio-copy.
- [x] Create GitHub Pages migration plan
**Done:** https://github.com/magento-research/pwa-studio/issues/1472#issuecomment-516498022.
- [x] Prepare a PR to redirect the old docs site to the new one
**Done:** https://github.com/magento-research/pwa-studio/pull/1494.
- [x] Ensure the `magento` organization has the `bundlesize`, `Danger-JS`, and `Now` Apps that the [magento-research](https://github.com/organizations/magento-research/settings/installations/) organization has
**Done:** It does not, but we can install them

## Move

- [x] Modify the http://pwastudio.io redirect to point to the new docs site location (@ericerway)
- [x] Rebuild / Deploy the docs site [from this repo](https://github.com/jcalcaben/migration-test/tree/gh-pages) (@jcalcaben)
- [x] Verify the http://pwastudio.io and https://magento-research.github.io/pwa-studio/ redirects are working (they should be `404`ing)
- [x] Update the `pwa-studio-copy` mirror with the latest changes
```
git fetch -p origin
git push --mirror
```
- [x] Move `pwa-studio` to `magento` organization on GitHub. [Instructions here](https://help.github.com/en/articles/transferring-a-repository#transferring-a-repository-owned-by-your-organization).
- [x] Rebuild, publish, and verify the docs site (https://magento.github.io/pwa-studio/)
- [x] Verify that [pwastudio.io]() and [the old docs site](https://magento-research.github.io/pwa-studio/) redirect to [the new docs site](https://magento.github.io/pwa-studio/)
- [x] Install [the required GitHub Apps](https://github.com/magento-research/pwa-studio/issues/1472#issuecomment-516566267)
- [x] Add the full list of outside collaborators to add to the Community Contributors team with write permissions
- [x] Notify the community (Slack, Community Sync), especially about the docs site move. Monitor Community Slack for problems
- [x] Merge #1488 
- [x] Merge [#14](https://github.com/magento/pwa-studio-cicd/pull/14) in `pwa-studio-cicd` repo

## After the Move

- [ ] Review the team's workflow to ensure that it is community oriented, make changes if necessary

### Owned by Us

- [x] Update repository references in `pwa-studio` markdown
**Done:** #1511 .
- [x] Update repository references in `pwa-studio` wiki 
- [x] Update repository references in `dev-docs`
- [x] Update repository references in `pwa-tests` repo
**Update:** none found
- [x] Update repository references in internal tools (Confluence, etc.)
**Created Issue:** #1524.
- [x] Delete https://github.com/magento-research/pwa-studio-copy repository
**Created Issue:** #1525.
- [x] Move other repositories to the `magento` organization:
  * `upward-php`
  * `magento2-upward-connector`
  * `pwa-tests`
  * `express-sharp`
  * `venia-consumer-example`

**Created Issue:** #1526.

#### Known Issues

- [x] Now.sh
**Issue Created:** #1519.
- [x] Coveralls is not reporting correctly
**Issue Created:** #1520.
- [x] CLA check
**Done:** Folks may have to re-sign the CLA in the `magento` org. Working for new contributors.
- [x] Old docs site `404`s

### Owned by EngCom

- [ ] Add `pwa-studio` as a pinned repository in `magento`
- [ ] Create GitHub project for `pwa-studio`
- [ ] Update EngCom's Portal tool
- [ ] Check contributions points mechanism
- [ ] Community Portal statistic. PWA project name
- [ ] Community Portal project card

**Comment by awilcoxa** [comment]
Are there any inbound/outbound links that we need to worry about breaking during the move? Docs, wiki, confluence, etc.?

**Comment by supernova-at** [comment]
> Are there any inbound/outbound links that we need to worry about breaking during the move? Docs, wiki, confluence, etc.?

We should definitely update those but GitHub will redirect for us in the meantime. The way I understand it, nothing will break. 🤞

**Comment by brendanfalkowski** [comment]
If we have a not-clone repo with a copy of all the code, and `magento-research/pwa-studio` is a remote that we forcefully pull from to get upstream changes...

As I understand it, that's how quite a few projects are keeping track and using PWA Studio today.

No idea if Git or GitHub gracefully handles that redirect. Probably not I'm guessing. Could be a confusing day weeks down the line if Git starts saying remote not found.

Not sure there's any way around it if the redirect doesn't apply.

**Comment by supernova-at** [comment]
> As I understand it, that's how quite a few projects are keeping track and using PWA Studio today.

Good point - @Jordaneisenburger that's how fallback studio works, right?

From what I understand, GitHub will redirect HTTP(S) and SSH requests to `magento-research/pwa-studio` to `magento/pwa-studio`. Those remotes should still be updated, but (at least in the short term) they will be redirected.

**Comment by supernova-at** [comment]
From https://help.github.com/en/articles/transferring-a-repository:

> All links to the previous repository location are automatically redirected to the new location. When you use git clone, git fetch, or git push on a transferred repository, these commands will redirect to the new repository location or URL. However, to avoid confusion, we strongly recommend updating any existing local clones to point to the new repository URL. You can do this by using git remote on the command line:

```
$ git remote set-url origin new_url
```

**Comment by Jordaneisenburger** [comment]
@supernova-at Fallback Studio is a full wrapper around the whole PWA Studio repo. It's used as a vendor and so Studio lives inside the fallback studio repo. Meaning that it wouldn't crash if the  Studio repo changed. I just have to update my upgrade script to get the latest Studio version from the new repo i think

**Comment by supernova-at** [comment]
Okay here is EngCom's checklist from previous repository transfers:

1. Create full repo copy
1. Check redirects: GitHub, Wiki, Git console
1. Check list of outside collaborators. Move them to Community Contributors team with write permissions
1. Move to Magento org
1. Create DevOps tickets to update Jenkins configuration to use new location
1. Check CLA service integration
1. Backlog GitHub app integration
1. EngCom portal tool Update
1. Check contributions points mechanism
1. Community portal statistic. MSI project name
1. Check release packaging builds
1. Check all documentations (DevDocs, Wiki)
1. Community Portal project card

Here are some thoughts / questions on the above:

> Create full repo copy

✅ Added to our checklist

Seems like a good idea "just in case". I don't think we want to officially [duplicate a repository](https://help.github.com/en/articles/duplicating-a-repository), but it would be smart to make sure we have a copy of the full repo _somewhere_ (even if it's just locally on someone's machine) before we initiate the transfer.

Questions:
1. What advantages does duplicating the repository have?
1. Am I correct in assuming we'd have the original `magento-research/pwa-studio` and the duplicated `magento-research/pwa-studio-new` and then we'd transfer `magento-research/pwa-studio-new` to `magento/pwa-studio`?
1. Where would that leave folks who fetch / pull / push to `magento-research/pwa-studio`? Would that still update the old repository?

> Check redirects: GitHub, Wiki, Git console

❌ Did not add to our checklist

If we don't do the full duplication we don't have to do this step, as the redirects won't kick in until we do the actual transfer.

> Check list of outside collaborators. Move them to Community Contributors team with write permissions

✅ Added to our checklist

I think we can get a list of folks who've contributed but where does that leave new contributors? Is there a process for them to get themselves added to Community Contributors team?

> Create DevOps tickets to update Jenkins configuration to use new location

❌ Did not add to our checklist

Not needed for PWA.

> EngCom portal tool Update

✅ Added to our checklist

I'm assuming `pwa-studio` will have to be added to EngCom's portal tool

> Check contributions points mechanism

✅ Added to our checklist

Questions:
1. How do we do this?

> Community portal statistic. PWA project name
> Community Portal project card

✅ Added to our checklist

I'm assuming EngCom will run point on these 😄 

> Check release packaging builds

❌ Did not add to our checklist

We should probably plan for a hiccup or two next time we use the automated release scripts. Updates can happen at that time.

**Comment by supernova-at** [comment]
Here is the full list of outside collaborators to add to the Community Contributors team:

0: {name: "zetlen", url: "https://github.com/zetlen"}
1: {name: "DrewML", url: "https://github.com/DrewML"}
2: {name: "jcalcaben", url: "https://github.com/jcalcaben"}
3: {name: "jimbo", url: "https://github.com/jimbo"}
4: {name: "supernova-at", url: "https://github.com/supernova-at"}
5: {name: "sirugh", url: "https://github.com/sirugh"}
6: {name: "Starotitorov", url: "https://github.com/Starotitorov"}
7: {name: "tjwiebell", url: "https://github.com/tjwiebell"}
8: {name: "mage2pratik", url: "https://github.com/mage2pratik"}
9: {name: "AlexeyKaryka", url: "https://github.com/AlexeyKaryka"}
10: {name: "sharkySharks", url: "https://github.com/sharkySharks"}
11: {name: "dpatil-magento", url: "https://github.com/dpatil-magento"}
12: {name: "codeAdrian", url: "https://github.com/codeAdrian"}
13: {name: "pcvonz", url: "https://github.com/pcvonz"}
14: {name: "vdiachenko", url: "https://github.com/vdiachenko"}
15: {name: "Serunde", url: "https://github.com/Serunde"}
16: {name: "JStein92", url: "https://github.com/JStein92"}
17: {name: "yogeshsuhagiya", url: "https://github.com/yogeshsuhagiya"}
18: {name: "vishal-7037", url: "https://github.com/vishal-7037"}
19: {name: "lewisvoncken", url: "https://github.com/lewisvoncken"}
20: {name: "rowan-m", url: "https://github.com/rowan-m"}
21: {name: "jahvi", url: "https://github.com/jahvi"}
22: {name: "ennostuurman", url: "https://github.com/ennostuurman"}
23: {name: "dani97", url: "https://github.com/dani97"}
24: {name: "brendanfalkowski", url: "https://github.com/brendanfalkowski"}
25: {name: "shakyShane", url: "https://github.com/shakyShane"}
26: {name: "rossmc", url: "https://github.com/rossmc"}
27: {name: "ronak2ram", url: "https://github.com/ronak2ram"}
28: {name: "real34", url: "https://github.com/real34"}
29: {name: "mtbottens", url: "https://github.com/mtbottens"}
30: {name: "mhhansen", url: "https://github.com/mhhansen"}
31: {name: "khoa-le", url: "https://github.com/khoa-le"}
32: {name: "jissereitsma", url: "https://github.com/jissereitsma"}
33: {name: "gil--", url: "https://github.com/gil--"}
34: {name: "belbiy", url: "https://github.com/belbiy"}
35: {name: "abrarpathan19", url: "https://github.com/abrarpathan19"}
36: {name: "VitaliyBoyko", url: "https://github.com/VitaliyBoyko"}
37: {name: "LucasCalazans", url: "https://github.com/LucasCalazans"}
38: {name: "Igloczek", url: "https://github.com/Igloczek"}
39: {name: "zengang", url: "https://github.com/zengang"}
40: {name: "zanilee", url: "https://github.com/zanilee"}
41: {name: "vitalics", url: "https://github.com/vitalics"}
42: {name: "sudeep-cedcoss", url: "https://github.com/sudeep-cedcoss"}
43: {name: "speedy008", url: "https://github.com/speedy008"}
44: {name: "shashidesilva", url: "https://github.com/shashidesilva"}
45: {name: "sanjay-wagento", url: "https://github.com/sanjay-wagento"}
46: {name: "realchriswells", url: "https://github.com/realchriswells"}
47: {name: "raith-hamzah", url: "https://github.com/raith-hamzah"}
48: {name: "prakashpatel07", url: "https://github.com/prakashpatel07"}
49: {name: "pradeep-cedcoss", url: "https://github.com/pradeep-cedcoss"}
50: {name: "philwinkle", url: "https://github.com/philwinkle"}
51: {name: "niklas-wolf", url: "https://github.com/niklas-wolf"}
52: {name: "neeta-wagento", url: "https://github.com/neeta-wagento"}
53: {name: "mzeis", url: "https://github.com/mzeis"}
54: {name: "mrtuvn", url: "https://github.com/mrtuvn"}
55: {name: "michaelyu0123", url: "https://github.com/michaelyu0123"}
56: {name: "matthewhaworth", url: "https://github.com/matthewhaworth"}
57: {name: "marcneubauer", url: "https://github.com/marcneubauer"}
58: {name: "marcin-piekarski", url: "https://github.com/marcin-piekarski"}
59: {name: "mageprince", url: "https://github.com/mageprince"}
60: {name: "kanhaiya5590", url: "https://github.com/kanhaiya5590"}
61: {name: "jflanaganuk", url: "https://github.com/jflanaganuk"}
62: {name: "jaimin-ktpl", url: "https://github.com/jaimin-ktpl"}
63: {name: "greenkeeper[bot]", url: "https://github.com/apps/greenkeeper"}
64: {name: "gavin2point0", url: "https://github.com/gavin2point0"}
65: {name: "fooman", url: "https://github.com/fooman"}
66: {name: "edwinbos", url: "https://github.com/edwinbos"}
67: {name: "davidverholen", url: "https://github.com/davidverholen"}
68: {name: "camdixon", url: "https://github.com/camdixon"}
69: {name: "bobmotor", url: "https://github.com/bobmotor"}
70: {name: "bobbyshaw", url: "https://github.com/bobbyshaw"}
71: {name: "bgkavinga", url: "https://github.com/bgkavinga"}
72: {name: "bbatsche", url: "https://github.com/bbatsche"}
73: {name: "artKozinets", url: "https://github.com/artKozinets"}
74: {name: "ankitsrivastavacedcoss", url: "https://github.com/ankitsrivastavacedcoss"}
75: {name: "andreas-ateles", url: "https://github.com/andreas-ateles"}
76: {name: "adrian-martinez-interactiv4", url: "https://github.com/adrian-martinez-interactiv4"}
77: {name: "PWAStudioBot", url: "https://github.com/PWAStudioBot"}
78: {name: "MarynaVozniuk", url: "https://github.com/MarynaVozniuk"}
79: {name: "Jakhotiya", url: "https://github.com/Jakhotiya"}

**Comment by VladimirZaets** [comment]
@supernova-at. 
About "Create full repo copy", we did it the first time, and yes, it's needed just to protect yourself from some extra cases. 
1. We did duplicate using [GitHub mirror approach](https://help.github.com/en/articles/duplicating-a-repository).
2. No, you should transfer origin repository, duplicate it's just back up copy for an extra case.
3. The  fetch / pull / push and other manipulation with "origin" will be automatically redirected to the new repository. ( You will see the notice in the terminal that remote repository migrated to another organization, but all functionality will work correctly) 


Take attention to project collaborators, because they permissions will not be transferred automatically.

```
Community portal statistic. PWA project name
Community Portal project card 
EngCom portal tool Update
``` 
-  we will check these points.

**Comment by supernova-at** [comment]
> Identify CICD integrations that may potentially break

Integrations:
  * dangerJS
  * now
  * others TBD

**Comment by supernova-at** [comment]
> Record project collaborators' permission set

| User | Access |
| --- | --- | 
| https://github.com/mehryar | Read |
| https://github.com/rowan-m | Read |
| https://github.com/tjwiebell | Admin |
| https://github.com/bbatsche | Admin |
| https://github.com/hrvojejurisic | Read |
| https://github.com/jahvi | Admin |
| https://github.com/antonkril | Admin |
| https://github.com/rganin | Admin |
| https://github.com/sirugh | Admin |
| https://github.com/maksek | Read |
| https://github.com/mhhansen | Admin |
| https://github.com/shakyShane | Read |
| https://github.com/zetlen | Admin |
| https://github.com/nuzil | Admin |
| https://github.com/ennostuurman | Read |
| https://github.com/jimbo | Admin |
| https://github.com/vrann | Admin |
| https://github.com/ericerway | Admin |
| https://github.com/jcalcaben | Admin |
| https://github.com/mmansoor-magento | Admin |
| https://github.com/DrewML| Admin |
| https://github.com/burtino| Admin |
| https://github.com/ishakhsuvarov | Admin |
| https://github.com/pcvonz | Admin |
| https://github.com/okorshenko | Admin |
| https://github.com/Crifen | Read |
| https://github.com/sharkySharks | Admin |
| https://github.com/vdiachenko | Read |
| https://github.com/melnikovi | Admin |
| https://github.com/okolesnyk | Read |

**Comment by supernova-at** [comment]
When we communicate the move to the community, this will be useful:

```
git remote set-url origin new_url
```

https://help.github.com/en/articles/changing-a-remotes-url

**Comment by supernova-at** [comment]
# GitHub Pages Migration Plan

## tl;dr

The existing docs site is going to `404` after the repository moves.
We have to be proactive about communicating that the docs site has moved because GitHub will not automatically redirect traffic to it.

## Overview / Context

Currently, our repository publishes to a GitHub Pages site.

It accomplishes this by maintaining a `gh-pages` branch. When this branch is edited, GitHub automatically updates https://magento-research.github.io/pwa-studio/.

We also have a custom domain http://pwastudio.io which redirects to https://magento-research.github.io/pwa-studio/. @ericerway owns http://pwastudio.io.

## The Problem

In [what's transferred with a repository](https://help.github.com/en/articles/transferring-a-repository#whats-transferred-with-a-repository), GitHub says:

> If the transferred repository contains a GitHub Pages site, then links to the Git repository on the Web and through Git activity are redirected. However, we don't redirect GitHub Pages associated with the repository.

When we move the `pwa-studio` repository to the `magento` organization, the `gh-pages` branch will move along with it. Any edits to the `gh-pages` branch after the move will result in https://magento.github.io/pwa-studio/ being updated, **not** https://magento-research.github.io/pwa-studio/.

Which means that anyone that has bookmarked https://magento-research.github.io/pwa-studio/ **will not** be redirected to the new https://magento.github.io/pwa-studio/.

Because http://pwastudio.io is a simple redirect, most / all bookmarks will directly point to https://magento-research.github.io/pwa-studio/, not http://pwastudio.io itself.

Further, once we move the repository and there no longer exists a repo at `magento-research/pwa-studio`, we anticipate that the existing https://magento-research.github.io/pwa-studio/ site will `404`.

Long story short, we will be leaving a lot of folks with a bookmark to a site that won't exist.

## Possible Solutions

### Wait

The idea here is that we put up a banner on the existing https://magento-research.github.io/pwa-studio/ site that lets visitors know that the docs site will be moving.

We'd need to decide how much time is "enough", but we wouldn't actually perform the move until the banner has been up for that amount of time.

### Communicate

This is the same as `Wait` except we only let the banner sit for a couple days. We'd have to be much more proactive about communicating because it would be more likely that fewer people would see the move notice.

Another downside is that anyone who has the docs site bookmarked but isn't in our Slack wouldn't be notified of the move.

This is the same outcome as if they didn't visit the site and read the banner, however.

### Re-create PWA Studio in magento-research

The repo would only contain the `gh-pages` branch which would keep up the banner that links visitors to the new docs site.

This would prevent https://magento-research.github.io/pwa-studio/ from `404`ing.

Unfortunately, this would probably also break any automatic redirecting that GitHub would do for us so that HTTP(S) / SSH traffic gets redirected to `magento/pwa-studio`. Therefore we do not think this solution is a good one.

**Comment by supernova-at** [comment]
# Installing Required GitHub Apps

## bundlesize

https://github.com/siddharthkp/bundlesize

## Danger-JS

https://github.com/apps/danger-js

## Now

https://zeit.co/github

**Comment by supernova-at** [comment]
Create ticket in JIRA - please check pwa-studio CLA integration

**Comment by brendanfalkowski** [comment]
Suggestion: after the move, add PWA Studio to the pinned repositories for the `magento` org (top middle slot looks good).

**Comment by supernova-at** [comment]
Closing. Move complete!
